pkg/operator/ceph/object/admin.go

```diff
@@ -108,7 +108,7 @@ var (
 
 // NewContext creates a new object store context.
 func NewContext(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo, name string) *Context {
-	return &Context{Context: context, Name: name, clusterInfo: clusterInfo}
+	return &Context{Context: context, Name: name, Realm: name, clusterInfo: clusterInfo}
```

Comment:
I don't see this added `Realm` parameter used anywhere. Did I miss something?

```suggestion
	return &Context{Context: context, Name: name, Realm: name, clusterInfo: clusterInfo}
```

Reply:
It is used when here https://github.com/rook/rook/blob/2b189ea2dece5aa2121beb614129e6812408f0b7/pkg/operator/ceph/object/admin.go#L249 when the `radosgw-admin` to get the user info (`GetUser()`) is run.
Otherwise only the "default" realm's user info is returned.

Reply:
I see. I think this approach may be missing something though. When an object store is set up for multisite with `CephObjectRealm`, `CephObjectZoneGroup`, and `CephObjectZone` resources, the realm is not equal to the name. Why not use `NewMultisiteContext()` here and use the realm provided from that? I think that would be the approach that allows this corner case to work as well.